test/byContract/registries/registries.test.ts

```diff
@@ -0,0 +1,475 @@
+
+            await datasetRegistryAsAdmin
+                .setName(ensRegistry.address, datasetRegistryEnsName)
+                .then((tx) => tx.wait());
+            expect(await reverseResolver.name(datasetRegistryNameHash)).to.equal(
+                datasetRegistryEnsName,
+            );
+
+            await workerpoolRegistryAsAdmin
+                .setName(ensRegistry.address, workerpoolRegistryEnsName)
+                .then((tx) => tx.wait());
+            expect(await reverseResolver.name(workerpoolRegistryNameHash)).to.equal(
+                workerpoolRegistryEnsName,
+            );
+        });
+
+        it('Should not set name when user is not the owner', async () => {
+            await expect(
+                appRegistry.setName(ensRegistry.address, 'new.app.registry.eth'),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+            await expect(
+                datasetRegistry.setName(ensRegistry.address, 'new.dataset.registry.eth'),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+            await expect(
+                workerpoolRegistry.setName(ensRegistry.address, 'new.workerpool.registry.eth'),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+        });
+    });
+
+    describe('setBaseURI', () => {
+        it('Should retrieve base URI', async () => {
+            const chainId = hre.network.config.chainId;
+            expect(await appRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/app/${chainId}/`,
+            );
+            expect(await datasetRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/dataset/${chainId}/`,
+            );
+            expect(await workerpoolRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/workerpool/${chainId}/`,
+            );
+        });
+
+        it('Should not set base URI when user is not the owner', async () => {
+            await expect(appRegistry.setBaseURI(`https://new.url.iex.ec/app/`)).to.be.revertedWith(
+                'Ownable: caller is not the owner',
+            );
+            await expect(
+                datasetRegistry.setBaseURI(`https://new.url.iex.ec/dataset/`),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+            await expect(
+                workerpoolRegistry.setBaseURI(`https://new.url.iex.ec/workerpool/`),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+        });
+    });
+
+    describe('App Registry', () => {
+        const createAppArgs = [
+            `App`,
+            'DOCKER',
+            constants.MULTIADDR_BYTES,
+            ethers.utils.id(`Content of my app`),
+            '0x1234',
+        ] as [string, string, BytesLike, BytesLike, BytesLike];
+        it('Should predict the correct address for future app creation', async () => {
+            const code = await appRegistry.proxyCode();
```

Comment:
I think coverage getters are missing for 
```
	address   public master;
	bytes     public proxyCode;
	bytes32   public proxyCodeHash;
	IRegistry public previous;
	bool      public initialized;
```
I primarily think about `Should get proxy code hash` where we should check `const proxyCodeHash = ethers.utils.keccak256(code);`

What do think guys (@zguesmi too)?